client/network/src/transport.rs

```diff
@@ -112,11 +129,23 @@ pub fn build_transport(
 		core::upgrade::SelectUpgrade::new(yamux_config, mplex_config)
 	};
 
-	let transport = transport
+	let tcp_transport = transport
 		.upgrade(upgrade::Version::V1Lazy)
 		.authenticate(authentication_config)
 		.multiplex(multiplexing_config)
-		.timeout(Duration::from_secs(20))
+		.timeout(Duration::from_secs(20));
+
+	let quic_transport = if enable_quic {
```

Comment:
wouldn't it be better if we return `tcp_transport` without an OptionalTransport when quic is disabled? I.e. why wrap tcp_transport in OrTransport if we know quic is None.

Reply:
Because it's an experimental feature. Once it is stable there will be OrTransport without optional quic. And stabilization will require very little amount of changes.

Reply:
IMHO either way is OK.